include/boost/astronomy/coordinate/arithmetic.hpp

```diff
@@ -72,54 +73,270 @@ auto cross
-        typename representation1_type::quantity3::unit_type()))*
+        bu::conversion_factor(typename cartesian1_type::quantity1::unit_type(),
+        typename cartesian1_type::quantity3::unit_type()))*
         (bg::get<2>(tempPoint2)*
-        bu::conversion_factor(typename representation2_type::quantity3::unit_type(),
-        typename representation2_type::quantity1::unit_type()))));
+        bu::conversion_factor(typename cartesian2_type::quantity3::unit_type(),
+        typename cartesian2_type::quantity1::unit_type()))));
 
     bg::set<2>(result, (bg::get<0>(tempPoint1)*bg::get<1>(tempPoint2)) -
         ((bg::get<1>(tempPoint1)*
-        bu::conversion_factor(typename representation1_type::quantity2::unit_type(),
-        typename representation1_type::quantity1::unit_type()))*
+        bu::conversion_factor(typename cartesian1_type::quantity2::unit_type(),
+        typename cartesian1_type::quantity1::unit_type()))*
         (bg::get<0>(tempPoint2)*
-        bu::conversion_factor(typename representation2_type::quantity1::unit_type(),
-        typename representation2_type::quantity2::unit_type()))));
+        bu::conversion_factor(typename cartesian2_type::quantity1::unit_type(),
+        typename cartesian2_type::quantity2::unit_type()))));
 
-    return Representation1
+
+    return 
+        cartesian_representation
         <
-            typename representation1_type::type,
+            typename cartesian1_type::type,
             bu::quantity<typename bu::multiply_typeof_helper
             <
-                typename representation1_type::quantity2::unit_type,
-                typename representation2_type::quantity3::unit_type>::type
+                typename cartesian1_type::quantity2::unit_type,
+                typename cartesian2_type::quantity3::unit_type>::type
             >,
             bu::quantity<typename bu::multiply_typeof_helper
             <
-                typename representation1_type::quantity3::unit_type,
-                typename representation2_type::quantity1::unit_type>::type
+                typename cartesian1_type::quantity3::unit_type,
+                typename cartesian2_type::quantity1::unit_type>::type
             >,
             bu::quantity<typename bu::multiply_typeof_helper
             <
-                typename representation1_type::quantity1::unit_type,
-                typename representation2_type::quantity2::unit_type>::type
+                typename cartesian1_type::quantity1::unit_type,
+                typename cartesian2_type::quantity2::unit_type>::type
             >
         >(result);
 }
 
+//!Returns the cross product of representation1(cartesian differential) and representation2
+template
+<
+    template<typename ...> class Representation2,
+    typename ...Args1,
+    typename ...Args2
+>
+auto cross
+(
+    cartesian_differential<Args1...> const& representation1,
+    Representation2<Args2...> const& representation2
```

Comment:
As you have made this function to support differentials, you should name the arguments differential1 and differential2 instead of representation1 and representation2 to maintain uniformity in the library. And change this whole function accordingly.

Reply:
ok. I'll do that.